src/app/app.tsx

```diff
@@ -226,3 +259,39 @@ function calculateNodeValue(data: RecursiveNode): number {
     );
   }
 }
+
+export function findNodeByVersion(
+  data: SunburstData | SunburstLeafNode | null,
+  version: string | null
+): SunburstData | SunburstLeafNode | null {
+  if (!data) {
+    return null;
+  }
+  if (version === null) {
+    return data;
+  }
+  if (isLeafNode(data)) {
+    if (data.name === version) {
+      return data;
+    } else {
+      return null;
+    }
+  }
+  if (data.name === version) {
+    return data;
+  } else if (!isLeafNode(data)) {
+    for (const child of data.children) {
+      const result = findNodeByVersion(child, version);
+      if (result) {
+        return result;
+      }
+    }
+  }
+  return null;
+}
+
+function isLeafNode(
+  node: SunburstData | SunburstLeafNode
+): node is SunburstLeafNode {
+  return (node as SunburstLeafNode)?.value != null;
+}
```

Comment:
You can import this from the d3-sunburst package - feel free to add it to sunburst/index

Reply:
👍